doc/dev/kdb-operations.md

```diff
@@ -61,13 +62,14 @@ Properties of `kdbGet()`:
 
 - After calling `kdbGet (kdb, ks, parentKey)`, the KeySet `ks` will contain _all keys_ (including their values) that are stored in _any backend_ with a mountpoint that is _below `parentKey`_.
 - After calling `kdbGet (kdb, ks, parentKey)`, below `parentKey` the KeySet `ks` will _only_ contain keys that are stored in a backend.
+  <!-- TODO: what about default:/ keys? Should they be kept to allow runtime defaults? -->
```

Comment:
input welcome

IMO it's cleaner, if `kdbGet` always reflects the KDB exactly. Defaults should be provided via the spec and if that isn't possible for some reason, you can always handle the `NULL` return value from `ksLookup`.

Reply:
default keys should be created by the spec plugin

Reply:
So this is the correct behaviour?

```c
Key * parentKey = keyNew ("/foo", KEY_END);
Key * x =  keyNew ("default:/foo/bar", KEY_END);
ksAppendKey (ks, x);
kdbGet (kdb, ks, parentKey);

// x is NEVER in ks, because x overlaps with the things loaded from the KDB
```

Reply:
Okay, but then the question is the same, but for `spec:/` and `proc:/`

If a `spec:/` backend exists at `spec:/foo` and a `proc:/` backend (e.g. for `poststorage` validation) exists at `proc:/foo`. Currently this would happen:

```c
Key * parentKey = keyNew ("/foo", KEY_END);
Key * x =  keyNew ("spec:/foo/bar", KEY_END);
ksAppendKey (ks, x);
Key * y =  keyNew ("proc:/foo/bar", KEY_END);
ksAppendKey (ks, y);
kdbGet (kdb, ks, parentKey);

// x and y are NEVER in ks, because x and y overlap with the things loaded from the KDB
// other Key * with the same name may exist in ks
```

So a hard-coded spec is possible, but you get either the entire hard-coded spec OR the entire spec from the KDB for one parent key of the spec. You can't have e.g. most of the spec from the KDB, but `spec:/foo/bar` is from the hard-coded spec.

Is that acceptable?

Reply:
Yes, I don't see a use case for individual hardcoded overrides. (Only the other way round I see a use case but even this case we want to eliminate, #4444). So the application either supplies full hardcoded proc or no proc and/or full hardcoded spec or no spec.